### PR TITLE
refactor: better giveaways

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: src/lib/database/migrations/1633201003755-V58_GiveawayAuthor.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { MigrationInterface, QueryRunner, TableColumn } from 'typeorm';
+
+export class V58GiveawayAuthor1633201003755 implements MigrationInterface {
+	public async up(queryRunner: QueryRunner): Promise<void> {
+		await queryRunner.addColumn('giveaway', new TableColumn({ name: 'author_id', type: 'varchar', length: '19', isNullable: true }));
```
Comment (favna) [inline src/lib/database/migrations/1633201003755-V58_GiveawayAuthor.ts:5]: Insert currently scheduled giveaways

**Review by favna — CHANGES_REQUESTED** [review]

File: src/lib/database/migrations/1633201003755-V58_GiveawayAuthor.ts (hunk 1 above)
Comment (favna) [reply src/lib/database/migrations/1633201003755-V58_GiveawayAuthor.ts:5]: Right now there is only 1:
```json
[
  {
    "id": 73766,
    "task_id": "delayedGiveawayCreate",
    "time": "2021-10-06 06:36:49.895000",
    "recurring": null,
    "catch_up": true,
    "data": "{\"title\": \"test\", \"endsAt\": 1633502270395, \"guildId\": \"872794074829176862\", \"minimum\": 1, \"channelId\": \"885472129850691584\", \"allowedRoles\": [], \"minimumWinners\": 1}"
  }
]
```

server data: https://discord.com/channels/541738403230777351/642137151626018818/893947227016409119

Comment (kyranet) [reply src/lib/database/migrations/1633201003755-V58_GiveawayAuthor.ts:5]: Unneeded, the lack of the property will make the constructor not assign `GiveawayEntity#authorId`, and so, it'll stay `null`, as current giveaways will.

**Comment by favna** [comment]
Functionally LGTM. Need to look into coverage targets & testing the code on Discord.

**Review by favna — APPROVED** [review]

**Pull request merged**